nexus/db-model/src/schema_versions.rs

```diff
@@ -0,0 +1,798 @@
+static KNOWN_VERSIONS: Lazy<Vec<KnownVersion>> = Lazy::new(|| {
+    vec![
+        // +- The next version goes here!  Duplicate this line, uncomment
+        // |  the *second* copy, then update that copy for your version,
+        // |  leaving the first copy as an example for the next person.
+        // v
+        // KnownVersion::new(next_int, "unique-dirname-with-the-sql-files"),
+        KnownVersion::new(45, "migration-demo"),
+        // The first many schema versions only vary by major or patch number and
+        // their path is predictable based on the version number.  (This was
+        // historically a problem because two pull requests both adding a new
+        // schema version might merge cleanly but produce an invalid result.)
+        KnownVersion::legacy(44, 0),
+        KnownVersion::legacy(43, 0),
+        KnownVersion::legacy(42, 0),
+        KnownVersion::legacy(41, 0),
+        KnownVersion::legacy(40, 0),
+        KnownVersion::legacy(39, 0),
+        KnownVersion::legacy(38, 0),
+        KnownVersion::legacy(37, 1),
+        KnownVersion::legacy(37, 0),
+        KnownVersion::legacy(36, 0),
+        KnownVersion::legacy(35, 0),
+        KnownVersion::legacy(34, 0),
+        KnownVersion::legacy(33, 1),
+        KnownVersion::legacy(33, 0),
+        KnownVersion::legacy(32, 0),
+        KnownVersion::legacy(31, 0),
+        KnownVersion::legacy(30, 0),
+        KnownVersion::legacy(29, 0),
+        KnownVersion::legacy(28, 0),
+        KnownVersion::legacy(27, 0),
+        KnownVersion::legacy(26, 0),
+        KnownVersion::legacy(25, 0),
+        KnownVersion::legacy(24, 0),
+        KnownVersion::legacy(23, 1),
+        KnownVersion::legacy(23, 0),
+        KnownVersion::legacy(22, 0),
+        KnownVersion::legacy(21, 0),
+        KnownVersion::legacy(20, 0),
+        KnownVersion::legacy(19, 0),
+        KnownVersion::legacy(18, 0),
+        KnownVersion::legacy(17, 0),
+        KnownVersion::legacy(16, 0),
+        KnownVersion::legacy(15, 0),
+        KnownVersion::legacy(14, 0),
+        KnownVersion::legacy(13, 0),
+        KnownVersion::legacy(12, 0),
+        KnownVersion::legacy(11, 0),
+        KnownVersion::legacy(10, 0),
+        KnownVersion::legacy(9, 0),
+        KnownVersion::legacy(8, 0),
+        KnownVersion::legacy(7, 0),
+        KnownVersion::legacy(6, 0),
+        KnownVersion::legacy(5, 0),
+        KnownVersion::legacy(4, 0),
+        KnownVersion::legacy(3, 3),
+        KnownVersion::legacy(3, 2),
+        KnownVersion::legacy(3, 1),
+        KnownVersion::legacy(3, 0),
+        KnownVersion::legacy(2, 0),
+        KnownVersion::legacy(1, 0),
+    ]
+});
+
+/// The earliest supported schema version.
+pub const EARLIEST_SUPPORTED_VERSION: SemverVersion =
+    SemverVersion::new(1, 0, 0);
+
+/// Describes one version of the database schema
+#[derive(Debug, Clone)]
+struct KnownVersion {
+    /// All versions have an associated SemVer.  We only use the major number.
+    semver: SemverVersion,
```

Comment:
> "We only use the major number"

This isn't *strictly* true, we do keep them ordered by minor/patch versions too, we just can't ensure backwards compatibility and recommend only bumping the major number.

I only bring this up because there already are legacy versions that only differ by minor versions.

Reply:
Sorry, what I meant was not "only the major number is ever non-zero" but rather "we only use the major number to determine compatibility".  I'll clarified this comment.